File: Partition Array According to Given Pivot/Partition_Array_According_to_Given_Pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        ArrayList<Integer> less = new ArrayList<>();
        ArrayList<Integer> equal = new ArrayList<>();
        ArrayList<Integer> greater = new ArrayList<>();
        int n = nums.length;
        for(int i = 0; i<n; i++) {
            if(nums[i] < pivot) {
                less.add(nums[i]);
            }
            else if(nums[i] > pivot) {
                greater.add(nums[i]);
            }
            else {
                equal.add(nums[i]);
            }
        }
        int eq = equal.size();
        int ls = less.size();
        int gr = greater.size();
        int ans[] = new int[n];
        int j = 0;
        for(int i = 0; i<ls; i++) {
            ans[j] = less.get(i);
            j++;
        }
        for(int i = 0; i<eq; i++) {
            ans[j] = equal.get(i);
            j++;
        }
        for(int i = 0; i<gr; i++) {
            ans[j] = greater.get(i);
            j++;
        }
        return ans;
    }
}

File: Partition Array According to Given Pivot/Partition_Array_According_to_Given_Pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        vector<int> less,equal,greater;
        int n = nums.size();
        for(int i = 0; i<n; i++) {
            if(nums[i] < pivot) {
                less.push_back(nums[i]);
            }
            else if(nums[i] > pivot) {
                greater.push_back(nums[i]);
            }
            else {
                equal.push_back(nums[i]);
            }
        }
        vector<int> ans;
        for(int i = 0; i<less.size(); i++) {
            ans.push_back(less[i]);
        }
        for(int i = 0; i<equal.size(); i++) {
            ans.push_back(equal[i]);
        }
        for(int i = 0; i<greater.size(); i++) {
            ans.push_back(greater[i]);
        }
        return ans;
    }
};